### ISSUE TITLE
Show .htaccess files on server

### ISSUE THREAD

**Issue by olivdee** [Natizyskunk/vscode-sftp#183]
Hi there, is there any chance we get the option to enable showing .htaccess files on the server?
Maybe something one could add inside **sftp.json**, like so:
```
{
...
"showHiddenFiles": true
...
}
```

Thanks in advance!

**Does this project help you?**
- [x] Yes. SFTP IS AWESOME!

**Comment by mavuriel** [comment]
I don't if that comes from server config, but I always see hidden files/folders in the sftp explorer and I can download it.🤔

![image](https://user-images.githubusercontent.com/46590763/187304858-e0080fb7-5818-4ef8-b51e-391449e33cdf.png)

**Comment by olivdee** [comment]
This is very strange, as I contacted the server support after your answer. They replied, that it cannot be a server issue, as:
a) there is no such setting in server/apache config that blocks or allows "invisible"/".dot" files like .htaccess
b) if this works with an FTP software (like FileZilla) then it should also work with another one as well (= YES it works with FileZilla on my side, I mean, FileZilla shows me the .htaccess file of the server/website whereas your SFTP extension does not show it on vscode...)

**Comment by Natizyskunk** [comment]
Are you sure you d'ont have a parameter in your vscode user config that could interfere ? As @mavuriel stated, the extension always show hidden files (and dotfiles) by default and i've implemented any option to not show them.

**Comment by godsdog** [comment]
Same issue here. Filezilla is showing .dot files, no .dot files in the explorer remote view.
The issue is only occuring on some servers?!

**Comment by Natizyskunk** [comment]
On my end everything work fine. All dotfiles are showing correctly.

**Comment by olivdee** [comment]
Sure. Which file is it?
Inside C:\\Users\[USERNAME]\AppData\Roaming\Code\User\settings.json there isn't much info (see below). Any other config file you have in mind?
`    "cSpell.enabled": false,
    "security.workspace.trust.untrustedFiles": "open",
    "sync.gist": "",
    "sync.autoDownload": true,
    "sync.autoUpload": true,
    "sync.forceUpload": true,
    "sync.forceDownload": true,
    "sync.quietSync": true,
    "window.commandCenter": true,
    "editor.wordWrap": "on"`

**Comment by Natizyskunk** [comment]
I was talinkg about the `sftp.json` config file in your `.vscode` folder inside your project.

In your user `settings.json` config file i can't see anything special. I suppose `sync` is another extension you're using.
Please try adding those two parameters inside your user config file and try again to see what's logged in the sftp debug pannel.

```json
"sftp.debug": true,
"sftp.printDebugLog": true,
```

**Comment by olivdee** [comment]
ah, I see...
this is the sftp.json of the project:
`{
    "name": "xxx.xxxxx.xxx",
    "host": "xxx.xxxxx.xxx",
    "protocol": "ftp",
    "port": 0,
    "username": "ftp@xxxxx.xxx",
    "password": "xxxxxxxxx",
    "remotePath": "/new2022/htdocs",
    "uploadOnSave": false
}` 

will add the two debug related ones and report back, thanks

**Comment by olivdee** [comment]
this is from the output tab right after inserting the two new lines:
`[09-30 20:39:07] [info] config at c:\git\neu.muenzankauf-muc.de {"remotePath":"/new2022/htdocs","uploadOnSave":false,"useTempFile":false,"openSsh":false,"downloadOnOpen":false,"ignore":[],"concurrency":4,"protocol":"ftp","connectTimeout":10000,"interactiveAuth":false,"secure":false,"remoteTimeOffsetInHours":0,"name":"xxx.xxxxx.xxx","host":"xxx.xxxxx.xxx","port":0,"username":"******","password":"******","sftp.debug":true,"sftp.printDebugLog":true}`
in the tab "debug console" it says "Please start a debug session to evaluate expressions" = do I need to do something here?

**Comment by Natizyskunk** [comment]
everything seems fine to me. This is inconsistent as some poeple are having the same problem but others don't have it and it doesn't seems to come from a specific configuration.
I'll keep an aeye on this and L'll keep you updated.

**Comment by olivdee** [comment]
do you think it makes any sense if I install a clean VS Code on the same machine with another user?

**Comment by kvd999** [comment]
i had the same problem, i changed the proftpd server config with ListOptions "-l" => ListOptions "-a", and the problem was solved, hope this help

**Comment by Natizyskunk** [comment]
For those using `proftpd`, please edit this config file `/etc/proftpd.conf`.
Search for the `ListOptions` parameter and change it from `"-l"` to `"-la"`.

**Comment by Natizyskunk** [comment]
Related issues : #61, #121

**Comment by olivdee** [comment]
there is no proftpd on the server as far as I can see - it is running CentOS Linux 7.9.2009 with Plesk Obsidian Version 18.0.47 Update #2 ... as I mentioned, using filezilla shows .htaccess files - do you think it is worth installing a clean vs code with your sftp extension on another user account on the same PC, or even on a total different PC and check again?

**Comment by NyaPPuu** [comment]
@olivdee @godsdog
![image](https://user-images.githubusercontent.com/1353557/194442798-adc684e5-39b8-445d-8a7d-938014c7183f.png)
Is this option checked in FileZilla?



I succeeded in reproducing the error.
FileZilla:
![image](https://user-images.githubusercontent.com/1353557/194442850-e548ffe0-8033-4c94-9bc4-54a768069f27.png)
vscode-sftp:
![image](https://user-images.githubusercontent.com/1353557/194442909-3a030590-953a-4652-8c5c-fd0aff5d012a.png)
If you check the options mentioned above, you can see hidden files in Filezilla, but not in sftp.
You can add a line from `vsftp.conf` to make hidden files visible to all clients.
`force_dot_files=YES`

**Comment by olivdee** [comment]
@NyaPPuu no, "Force show hidden files" is not checked with my FileZilla, but .htaccess are showing there! Checked on 2 different online servers with different setup/OS...

since there is no /etc/vsftpd.conf, what should/could I do?

**Comment by olivdee** [comment]
what is proftpd and where is the config? not using such an extension...On Oct 3, 2022 15:51, kvd999 ***@***.***> wrote:
i had the same problem, i changed the proftpd server config with ListOptions "-l" => ListOptions "-a", and the problem was solved, hope this help

**Comment by janis-ps** [comment]
I discovered that the -la switch to FTP list command was removed here due to partial support: https://github.com/liximomo/vscode-sftp/commit/c2608f7573e8169fae1e72b6e254ea7ee292dfa3

Maybe it could be re-added as an option?

**Comment by lucaele** [comment]
Any fix for show dot files like .env?
I'm using SFTP ver 1.16.1

**Comment by Natizyskunk** [comment]
@lucaele,
Yes that's because when you're using the sftp protocol, we set the parameter `showHiddenFiles` to TRUE in the `list()` command but when using the ftp protocol, we set the parameter `showHiddenFiles` to FALSE in the `list()` command that has the effect to remove the `-al` options from the ftp `LIST` command.

@janis-ps,
Thanks, I will take a look at it more in depth.

**Comment by rodgersl** [comment]
hey @Natizyskunk just wanted to check if you had made any progress on this issue?

I have a similar issue with a CentOS Plesk server, connecting with this extension I can't see hidden / dot files but if I connect with another FTP client I can see them no problem.

Thanks for your work on this extension, it's really appreciated 😀

**Comment by giuliopons** [comment]
Hi, same problem. Can I have protocol set to `ftp` and manually force the `showHiddenFiles` to `true` in `sftp.json`?

**Comment by Nalik27** [comment]
Also faced this problem.
When I use the SFTP protocol, then I see the .htaccess file
When I use the FTP protocol, I don't see the .htaccess file

Can you please tell me how to enable .htaccess display?
Can this be done in the sftp.json settings?

**Comment by giuliopons** [comment]
> Also faced this problem. When I use the SFTP protocol, then I see the .htaccess file When I use the FTP protocol, I don't see the .htaccess file
> 
> Can you please tell me how to enable .htaccess display? Can this be done in the sftp.json settings?

I've asked my provider to switch to sftp, and solved the problem.

**Comment by Nalik27** [comment]
> > Also faced this problem. When I use the SFTP protocol, then I see the .htaccess file When I use the FTP protocol, I don't see the .htaccess file
> > Can you please tell me how to enable .htaccess display? Can this be done in the sftp.json settings?
> 
> I've asked my provider to switch to sftp, and solved the problem.

Thanks for your willingness to help!
My hoster says there is no way to provide me with SFTP. Because FTP is my only option.

**Comment by BigBlockStudios** [comment]
So why can't you just add a config option allowing us to modify the LIST command on a per instance basis?  Best option would be in SFTP config if possible (so we can configure it for multiple connections or profiles)

Changing Proftpd server configs, adding files in /etc/ is not a valid solution as not everyone will have access to do that nor does every service use proftpd! 
Not every service runs SFTP either ....  this problem has been around for years.

**Comment by hlaurent26** [comment]
I can confirm that. When I connect using the SFTP protocol, I can see the hidden files.
